gdal/frmts/raw/byndataset.cpp

```diff
@@ -0,0 +1,766 @@
+    memcpy( &poDS->hHeader.nDLat,      poOpenInfo->pabyHeader + 16, 2 );
+    memcpy( &poDS->hHeader.nDLon,      poOpenInfo->pabyHeader + 18, 2 );
+    memcpy( &poDS->hHeader.nGlobal,    poOpenInfo->pabyHeader + 20, 2 );
+    memcpy( &poDS->hHeader.nType,      poOpenInfo->pabyHeader + 22, 2 );
+    memcpy( &poDS->hHeader.dfFactor,   poOpenInfo->pabyHeader + 24, 8 );
+    memcpy( &poDS->hHeader.nSizeOf,    poOpenInfo->pabyHeader + 32, 2 );
+    memcpy( &poDS->hHeader.nVDatum,    poOpenInfo->pabyHeader + 34, 2 );
+    memcpy( &poDS->hHeader.nDescrip,   poOpenInfo->pabyHeader + 40, 2 );
+    memcpy( &poDS->hHeader.nSubType,   poOpenInfo->pabyHeader + 42, 2 );
+    memcpy( &poDS->hHeader.nDatum,     poOpenInfo->pabyHeader + 44, 2 );
+    memcpy( &poDS->hHeader.nEllipsoid, poOpenInfo->pabyHeader + 46, 2 );
+    memcpy( &poDS->hHeader.nByteOrder, poOpenInfo->pabyHeader + 48, 2 );
+    memcpy( &poDS->hHeader.nScale,     poOpenInfo->pabyHeader + 50, 2 );
+    memcpy( &poDS->hHeader.dfWo,       poOpenInfo->pabyHeader + 52, 8 );
+    memcpy( &poDS->hHeader.dfGM,       poOpenInfo->pabyHeader + 60, 8 );
+    memcpy( &poDS->hHeader.nTideSys,   poOpenInfo->pabyHeader + 68, 2 );
+    memcpy( &poDS->hHeader.nRealiz,    poOpenInfo->pabyHeader + 70, 2 );
+    memcpy( &poDS->hHeader.dEpoch,     poOpenInfo->pabyHeader + 72, 4 );
+    memcpy( &poDS->hHeader.nPtType,    poOpenInfo->pabyHeader + 76, 2 );
+
+    /********************************/
+    /* Scale boundaries and spacing */
+    /********************************/
+
+    double dfSouth = (double) poDS->hHeader.nSouth;
+    double dfNorth = (double) poDS->hHeader.nNorth;
+    double dfWest  = (double) poDS->hHeader.nWest;
+    double dfEast  = (double) poDS->hHeader.nEast;
+    double dfDLat  = (double) poDS->hHeader.nDLat;
+    double dfDLon  = (double) poDS->hHeader.nDLon;
+
+    if( poDS->hHeader.nScale == 1 ) 
+    {
+        dfSouth /= 1000;
+        dfNorth /= 1000;
+        dfWest  /= 1000;
+        dfEast  /= 1000;
+        dfDLat  /= 1000;
+        dfDLon  /= 1000;
+    }
+
+    /******************************/
+    /* Calculate rows and columns */
+    /******************************/
+
+    poDS->nRasterYSize = (GInt32) ( ( ( dfNorth - dfSouth ) / dfDLat ) + 1.0 );
+    poDS->nRasterXSize = (GInt32) ( ( ( dfEast  - dfWest  ) / dfDLon ) + 1.0 );
+
+    if (!GDALCheckDatasetDimensions(poDS->nRasterXSize, poDS->nRasterYSize))
+    {
+        delete poDS;
+        return nullptr;
+    }
+
+    /*****************************/
+    /* Build GeoTransform matrix */
+    /*****************************/
+
+    poDS->adfGeoTransform[0] = ( dfWest - ( dfDLon / 2.0 ) ) / 3600.0;
+    poDS->adfGeoTransform[1] = dfDLon / 3600.0;
+    poDS->adfGeoTransform[2] = 0.0;
+    poDS->adfGeoTransform[3] = ( dfNorth + ( dfDLat / 2.0 ) ) / 3600.0;
+    poDS->adfGeoTransform[4] = 0.0;
+    poDS->adfGeoTransform[5] = -1 * dfDLat / 3600.0;
+
+    CPLDebug("BYN","South           = %d",poDS->hHeader.nSouth);
```

Comment:
protect all those verbose debug info by #ifdef DEBUG so they don't appear in non-DEBUG builds